<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="top_layout">
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="leftMargin">
        <number>9</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>0</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>ADC Settings</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_4">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Sample Rate</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>PGA Setting</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Input Buffer</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="srate_dropdown">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Set the DStat analog to digital converter sample rate. Higher sample rates have increased noise but higher temporal resolution. Lower sample rates have decreased noise but lower temporal resolution. Selections that are multiples of 60 Hz are recommended to reduce interference due to mains hum.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="ibuffer_dropdown"/>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="pga_dropdown">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Adjust programmable gain amplifer built into the DStat's analog to digital converter.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Potentiostat Settings</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_5">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="1">
              <widget class="QComboBox" name="gain_dropdown">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Adjust the gain of the pre-amp connected to the analog to digital converter. Full scale (FS) current is listed. Select a setting with appropriate headroom to avoid clipping.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Gain</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Experiment</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_exp">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="3" column="0">
              <widget class="QFrame" name="frame_potentiometry">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QGridLayout" name="gridLayout_5">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item row="0" column="1">
                 <widget class="QDoubleSpinBox" name="pot_time">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Seconds to run the experiment. Maximum value is 65535 or enter 0 to run experiment forever until &amp;quot;Stop&amp;quot; button is clicked.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="decimals">
                   <number>0</number>
                  </property>
                  <property name="maximum">
                   <double>65535.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Time [s]:</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QFrame" name="frame_LSV">
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item alignment="Qt::AlignHCenter">
                 <widget class="QLabel" name="label_19">
                  <property name="text">
                   <string>Preconditioning</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="Line" name="line">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frame_9">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_7">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <item row="1" column="1">
                    <widget class="QDoubleSpinBox" name="lsv_vclean">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="2">
                    <widget class="QDoubleSpinBox" name="lsv_tclean">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1" alignment="Qt::AlignHCenter">
                    <widget class="QLabel" name="label_12">
                     <property name="text">
                      <string>Potential [mV]</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="0">
                    <widget class="QLabel" name="label_15">
                     <property name="text">
                      <string>Deposition</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="2" alignment="Qt::AlignHCenter">
                    <widget class="QLabel" name="label_13">
                     <property name="text">
                      <string>Time [s]</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="label_14">
                     <property name="text">
                      <string>Cleaning</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="2">
                    <widget class="QDoubleSpinBox" name="lsv_tdep">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="1">
                    <widget class="QDoubleSpinBox" name="lsv_vdep">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                     <property name="minimum">
                      <double>-5000.000000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>0.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item alignment="Qt::AlignHCenter">
                 <widget class="QLabel" name="label_20">
                  <property name="text">
                   <string>Experiment Settings</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="Line" name="line_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frame_10">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_8">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <property name="horizontalSpacing">
                    <number>6</number>
                   </property>
                   <item row="0" column="0">
                    <widget class="QLabel" name="label_8">
                     <property name="text">
                      <string>Start [mV]</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="lsv_start_mV">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="label_16">
                     <property name="text">
                      <string>Stop [mV]</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="0">
                    <widget class="QLabel" name="label_17">
                     <property name="text">
                      <string>Slope [mV/s]</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1">
                    <widget class="QDoubleSpinBox" name="lsv_stop_mV">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="1">
                    <widget class="QDoubleSpinBox" name="lsv_slope">
                     <property name="decimals">
                      <number>0</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QFrame" name="Type">
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QGridLayout" name="gridLayout_6">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <property name="spacing">
                 <number>0</number>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>Type</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QComboBox" name="exptype_dropdown">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_7">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="execute_btn">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>50</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Start the DStat experiment.</string>
               </property>
               <property name="text">
                <string>Execute</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="stop_btn">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>50</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Stop the current experiment running on the DStat. Collected data will be saved.</string>
               </property>
               <property name="text">
                <string>Stop</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="status_box">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>Status messages will appear here...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>USB Connection</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_8">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="1">
              <widget class="QComboBox" name="port_dropdown"/>
             </item>
             <item row="0" column="3">
              <widget class="QPushButton" name="connect_btn">
               <property name="text">
                <string>Connect</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="refresh_btn">
               <property name="text">
                <string>Refresh</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame_2">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="PlotWidget" name="exp_plot">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
